### PR TITLE
Add previous version compatibility test

### PR DESCRIPTION
**Pull request by laodouya**
1. Add old bp/miner/client compatibility test
2. Use common volumes in host for build bins

### CODE REVIEW

File: test/compatibility/specific_old.sh [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    *)
+        exit 1
+        ;;
+esac
+
+
+# start bp
+nohup ${BPBIN} -config ${PROJECT_DIR}/test/integration/node_0/config.yaml 2>${LOGS_DIR}/bp0.log &
+nohup ${BPBIN} -config ${PROJECT_DIR}/test/integration/node_1/config.yaml 2>${LOGS_DIR}/bp1.log &
+nohup ${BPBIN} -config ${PROJECT_DIR}/test/integration/node_2/config.yaml 2>${LOGS_DIR}/bp2.log &
```
Comment (leventeliu) [inline test/compatibility/specific_old.sh]: nohup xxx >xxx.log 2>&1 &

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -23,12 +25,22 @@ before_script:
...
 #  - curl -fSL https://github.com/haya14busa/reviewdog/releases/download/$REVIEWDOG_VERSION/reviewdog_linux_amd64 -o ~/bin/reviewdog && chmod +x ~/bin/reviewdog
 
-test-my-project:
+build:
+  stage: build
+  script:
+    - make clean
+    - make use_all_cores
+    - rm /CovenantSQL_bins/current/*
+    - cp bin/* /CovenantSQL_bins/current/
```
Comment (xq262144) [inline .gitlab-ci.yml]: Potential cross-interference of multiple `gitlab-ci` tasks.

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -1,10 +1,20 @@
...
+  - test
+  - cleanup
+
 variables:
   REVIEWDOG_VERSION: 0.9.11
   REVIEWDOG_GITLAB_API_TOKEN: $REVIEWDOG_TOKEN
   CODECOV_TOKEN: $CODECOV_TOKEN
   UNITTESTTAGS: linux sqlite_omit_load_extension
+  CACHE_DIR: /CovenantSQL_bins
+  PIPLINE_CACHE: $CACHE_DIR/$CI_PIPELINE_IID
```
Comment (xq262144) [inline .gitlab-ci.yml]: Typo here, PIPLINE_CACHE -> PIPELINE_CACHE

**Review by xq262144 — APPROVED** [review]
Approve

**Review by leventeliu — APPROVED** [review]
LGTM.

**Pull request merged**